lisp/ess-custom.el

```diff
@@ -1282,11 +1287,12 @@ global) values from ESS indentation variables."
                  (const RStudio)
                  (const RStudio-)
                  (const DEFAULT))
-  :group 'ess-edit)
+  :group 'ess-edit
+  :safe #'ess-style-safe)
 
-;; the real setting of this happens via <foo>-editing-alist:
-(defvar ess-style ess-default-style
-  "Current ESS indentation style, see `ess-style-alist' for more.")
+(defun ess-style-safe (v)
+  "Return non-nil if V is a style in `ess-style-alist'."
+  (member v (mapcar 'car ess-style-alist)))
```

Comment:
Are you sure this one doesn't break when a user adds a personal style to ess-style-alist?

Reply:
It appears to work fine. I'll add some ERT tests and push them in a minute.

Reply:
OK, done

Reply:
> How about just safing it to a symbolp?

That works, I guess. No need to be clever here. Done.

Reply:
I think we need to autoload it still, otherwise people with melpa might see the warning depending on how .dir-locals is configured.

Reply:
OK, done. I think I addressed all your feedback, so I'll merge soonish unless I hear otherwise.